### ISSUE TITLE
Unblock assistant button & simplify opting out after license has expired 

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/quickFilters#167]
Currently the assistant button is used to tell quickFilters Pro users about an expired license:

![image](https://user-images.githubusercontent.com/6314674/226105568-ff9301c9-28bf-4baa-a0b5-d01bc3e28729.png)

Unfortunately this also blocks access to the assistant feature (unless the license is either removed or renewed). Instead clicking the button goes straight to the license tab for easy renewal:

![image](https://user-images.githubusercontent.com/6314674/226105594-50824d06-aeb9-4686-96b9-1a1a90b56a6b.png)

This also means that using the assistant is effectively blocked for licensed users after expiry. It definitely needs to be fixed.

- Removing the license unblocks the assistant feature (through using the button). This should be made more obvious
- Give the user a chance to use the assistant feature even if the license is expired. At the moment it only redirects to the license page of options and doesn't do anything else.

**Comment by RealRaven2000** [comment]
When the user hovers on the "License Expired" button there is a tooltip explaining that other options can be opened via right-click. 

![image](https://user-images.githubusercontent.com/6314674/226125717-2dba5fdd-3e89-4966-bb35-6af7601bffeb.png)

It might be a good idea to open a dropdown here  which can activate / deactivate the assistant (and may give access to all other pages on the quickFilters settings), like this:

![image](https://user-images.githubusercontent.com/6314674/226126838-03656f7d-a550-4588-a228-f7317b9d8d71.png)

I think it might even be better to always show this menu (with all options) when this button is clicked. As the first option (Activate Assistant) is very close to the button it would still be easy enough to discover / click.

**Comment by RealRaven2000** [comment]
An additional thing that could help with understanding / avoiding any restrictions / undesired behavior of the Add-on would be to add a button [_remove license..._] if the license is expired:

![image](https://user-images.githubusercontent.com/6314674/226129627-ac7052a3-f397-411e-992e-ceef42fb5888.png)

The elipsis ... would denote that this is not an immediate action and it should display a message explaining something along these lines:

"If your license is expired you may want to remove it from quickFilters to return the Add-on to its initial behavior. Are you sure you want to remove the license? 

You can always enter it again to be eligible for renewal or buy a new one later."

[Ok] [Cancel]

**Comment by galacticEngineer** [comment]
This seems like a good solution, but I'm not sure I'd feel comfortable clicking the remove license button, even with the ellipsis, unless I understood in advance that the purpose was to return to the original non-licensed functionality. I wonder if it wouldn't be better to simply remove the license automatically when it expires and revert to non-licensed functionality. If the user tries to use a function requiring a license, then they could be informed that their license had expired.

**Comment by RealRaven2000** [comment]
> This seems like a good solution, but I'm not sure I'd feel comfortable clicking the remove license button, even with the ellipsis, unless I understood in advance that the purpose was to return to the original non-licensed functionality. 

How could I improve the sentence "remove it from quickFilters to return the Add-on to its initial behavior. " I am trying not to be too long winded here because people do not want to read a load of text and I have to translate this into 18 more languages.

> I wonder if it wouldn't be better to simply remove the license automatically when it expires and revert to non-licensed functionality. If the user tries to use a function requiring a license, then they could be informed that their license had expired.

No if I remove the license automatically completely you will lose between 30-60% money off that you save when you renew your license (and in the case of SmartTemplates / QuickFolders you can also use an expired Standard license to upgrade to Pro and get a full year, so you are often saving 50% or more)

**Comment by galacticEngineer** [comment]
Sorry, I was misinterpreting the text you proposed. I didn't realize that all of that text would be displayed to the user and was thinking that only the text highlighted in yellow would show. WIth the additional text, the meaning is clear.

With respect to not wanting to automatically remove the license due to the loss of a discount, that's a different problem that I hadn't realized. Given that removing the license takes away the opportunity for a discount, it seems like that should also be made clear in the text that explains the purpose of removing the license.  Ideally, I think that the discount offer would be handled separately from the restoration of normal non-licensed functionality. There should be a way to remember that the user had been licensed and was eligible for a discount independent of having the license key installed, or alternatively, and probably easier, a way to restore non-licensed functionality without removing the license key such as a simple checkbox to say, "I know my license has expired, but I just want the normal functionality for now."

**Comment by RealRaven2000** [comment]
So here is a first test version that implements some of the ideas discussed. Instead of having a simply pushbutton that has potentially multiple uses, I have turned this into a dropdown menu, with the first menu item giving the option to toggle (start / stop) the assistant. [I still need to adjust the text for toggling it off].

![image](https://user-images.githubusercontent.com/6314674/226186556-bb9f0893-3ec9-4eab-9141-09b146fce399.png)

If the license is expired, there will be an additional menu item at the top, in order to check the current license:

![image](https://user-images.githubusercontent.com/6314674/226186664-f99d7329-a9ee-422a-bb50-e155dca5ff7b.png)

The menu also gives access to the command for running the filters (on folder or on selected mails). 
It also can show the settings menu and the latest news (splash screen).
As last thing I am contemplating adding a menu item (get quickFilters Pro...) at the very end for non-licensed users. 

But this definitely already feels a lot more functional as you are now not locked out of the functionality even with an expired license.


[quickFilters-wx-5.8.2pre9.zip](https://github.com/RealRaven2000/quickFilters/files/11011499/quickFilters-wx-5.8.2pre9.zip)

To install this version, simply download the zip file and then drag the file into Thunderbird Add-ons Manager (without extracting contents).

**Comment by RealRaven2000** [comment]
Did some more work on the menu:

- fixed the 1st menu item "check license status..." to go to the license dialog panel. It will be removed automatically after license is renewed.
- start quickFilters Assistant will now work immediately (even if news were ignored) - I am still thinking about whether and how I could convince people to read the version log because I feel it is important to know how the software behaves differently
- fixed "filter selected emails" - To Do: change label to something more descriptive
- Fixed styles to be more cohesive
- Added item "get quickFilters Pro" for users without license, this opens the license choice dialog
- To Do: translate the new strings "Check license status…" and "Latest news on quickFilters…"

[quickFilters-wx-5.8.2pre29.zip](https://github.com/RealRaven2000/quickFilters/files/11011929/quickFilters-wx-5.8.2pre29.zip)

To install this version, simply download the zip file and then drag the file into Thunderbird Add-ons Manager (without extracting contents).

**Comment by RealRaven2000** [comment]
Tidied up the menu and added a Tools submenu:

![image](https://user-images.githubusercontent.com/6314674/226772773-fb6e68f6-c2bf-41a3-860f-47a26fc61ceb.png)

This makes it also easier to find these commands:
- Message Filters - opens the dialog that lists the filters
- Create Filter from Message - same as the command from the context menu in the list of mails.
- Find filters - searches the filters that affect (move mail to) the current folder

[quickFilters-wx-5.9pre50.zip](https://github.com/RealRaven2000/quickFilters/files/11035185/quickFilters-wx-5.9pre50.zip)

To do: localize the label "Tools"

**Comment by RealRaven2000** [comment]
Unblocking the assistant after license expired was implemented in 5.9 - released on 28/03/2023.
I have added a new issue #176 for opting out of the license after expiry.
